module/zfs/arc.c

```diff
@@ -9112,7 +9113,7 @@ l2arc_apply_transforms(spa_t *spa, arc_buf_hdr_t *hdr, uint64_t asize,
 		if (psize < asize)
 			memset((char *)tmp + psize, 0, asize - psize);
```

Comment:
I think this might need to change if asize is not always >= bufsize, no? In particular, I think in Linux if you turn off the init_on_alloc behavior, this might result in getting nonsense back in the delta between asize and size...

Reply:
Do we care about that region, though? We're not going to write anything beyond `asize` out to the disk, since that's the physical allocation size.

Reply:
I don't know of any case that we do, but I also couldn't think of a case when asize would have been larger than size or psize when I wrote the first patch, and I'd be surprised if the delta here was a significant performance hit compared to if something in the future stacks on this and accesses past the end for some reason.